pkg/reconciler/pipelinerun/resources/validate_params_test.go

```diff
@@ -121,3 +121,83 @@ func TestValidateParamTypesMatching_Invalid(t *testing.T) {
 		})
 	}
 }
+
+func TestValidateExpectedParametersProvided_Valid(t *testing.T) {
+	tcs := []struct {
+		name          string
+		p             *v1beta1.Pipeline
+		pr            *v1beta1.PipelineRun
+		errorExpected bool
```

Comment:
i highly recommend that instead of using a flag like this and mixing valid and invalid test cases, instead make two completely different tests, one with all the valid cases and one with all the invalid cases

its way easier to maintain over time and understand whats covered vs whats not, and what's valid vs what's not. it also avoids introducing bugs in the test itself around handling the flag

Reply:
it actually looks like you did that down below already so you can remove the flag from this test probably :D (as long as the failure case in this test is covered in the other test also)